### DISCUSSION TITLE
Does system include extraction require a file extension (to infer type) ?

### DISCUSSION THREAD

**Discussion by mellery451** [clangd/clangd#1849] [Q&A] [answered]
I have source files with no extension (I'm looking at you Eigen: https://gitlab.com/libeigen/eigen/-/tree/master/Eigen?ref_type=heads)  and I get errors like this:

      System include extraction: invalid file type for 

Does the std lib extraction require an extension? It already has the tool (compiler ) in the command ..not sure why extension matters? Is there a way to specify a default extension to infer if none exists?

**Comment by sam-mccall** [comment, marked as answer]
Type is definitely needed, because it's usual for clang/gcc to have different header search paths for C vs C++, for example.

The language can be specified either via flags (e.g. `-xc++-header`) or guessed based on extension. You hit that log line if neither is present, so we can't get the type.

But in that case I'm curious how the file gets parsed at all, and what the compile command actually is. The command `clang++ -fsyntax-only some/file/without/extension` is not valid for parsing - the clang driver treats the file as a linker input, not a source file.

**Comment by mellery451** [comment]
it was missing -x from these files...adding it indeed helps. thanks for the pointer
